File: app/src/main/java/abdullah/elamien/bloodbank/models/Donor.java
package abdullah.elamien.bloodbank.models;

import android.widget.ImageView;

import androidx.databinding.BindingAdapter;

import com.bumptech.glide.Glide;

public class Donor {

    private String name;
    private String phone;
    private String image_;
    private String address;
    private String bloodType;
    private String bio;
    private String userId;
    private int age;

    public Donor() {
    }

    @BindingAdapter("donorImage")
    public static void loadDonorImage(ImageView view, String imageUrl) {
        Glide.with(view.getContext())
                .load(imageUrl)
                .into(view);
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getImage_() {
        return image_;
    }

    public void setImage_(String image_) {
        this.image_ = image_;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getBloodType() {
        return bloodType;
    }

    public void setBloodType(String bloodType) {
        this.bloodType = bloodType;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public String getBio() {
        return bio;
    }

    public void setBio(String bio) {
        this.bio = bio;
    }
}

File: app/src/main/java/abdullah/elamien/bloodbank/models/Donor.kt
package abdullah.elamien.bloodbank.models

import android.widget.ImageView

import androidx.databinding.BindingAdapter

import com.bumptech.glide.Glide

class Donor {

    var name: String? = null
    var phone: String? = null
    var image_: String? = null
    var address: String? = null
    var bloodType: String? = null
    var bio: String? = null
    var userId: String? = null
    var age: Int = 0

    companion object {

        @BindingAdapter("donorImage")
        @JvmStatic
        fun loadDonorImage(view: ImageView, imageUrl: String) {
            Glide.with(view.context)
                    .load(imageUrl)
                    .into(view)
        }
    }
}
